From 2de4492754e1f86c29c20d8d5285a96647a5ea8b Mon Sep 17 00:00:00 2001
From: Suman Anna <s-anna@ti.com>
Date: Wed, 1 Feb 2017 13:37:58 -0600
Subject: [PATCH 189/205] ARM: dts: AM33xx: Add the PRUSS DT nodes

Add the DT nodes for the PRU-ICSS on AM33xx family of SoCs.
The AM33xx SoCs contain a single PRU-ICSS instance and is
represented by the pruss-soc-bus node and a child PRUSS node.
PRU-ICSS is not supported on AM3352 SoC though in the AM33xx
family, so the nodes are added in disabled state to the common
am33xx dtsi file. They should be enabled only for boards that
use a SoC containing PRU-ICSS in the derivative board files.

The DT nodes use all standard properties. The regs property
defines the addresses for various modules within the PRUSS,
including all the Data RAMs, the Instruction RAM per core,
the Industrial Ethernet Peripheral (IEP), the Real Time Media
Independent Interface controller (MII_RT), CFG, Debug and
Control sub-modules.

The two PRU cores within the PRUSS and the interrupt controller
are represented as child nodes (so platform devices themselves)
of the PRUSS parent node. The default support for virtio/rpmsg
communication between MPU and a PRU core is provided through
PRU system events instead of OMAP mailboxes. This is the
preferred approach as it eliminates an external peripheral
access from the PRU-side. The difference from MPU would be
minimal in using one versus the other.

Signed-off-by: Suman Anna <s-anna@ti.com>
Signed-off-by: Andrew F. Davis <afd@ti.com>
Signed-off-by: Keerthy <j-keerthy@ti.com>
(cherry picked from commit 29a92fae0104aae452f8af4e94dd29f32db4ea24)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm/boot/dts/am33xx.dtsi |   63 +++++++++++++++++++++++++++++++++++++++++
 1 file changed, 63 insertions(+)

diff --git a/arch/arm/boot/dts/am33xx.dtsi b/arch/arm/boot/dts/am33xx.dtsi
index 8022d2b..1fde2136 100644
--- a/arch/arm/boot/dts/am33xx.dtsi
+++ b/arch/arm/boot/dts/am33xx.dtsi
@@ -941,6 +941,69 @@
 			};
 		};
 
+		pruss_soc_bus: pruss_soc_bus@4a326000 {
+			compatible = "ti,am3356-pruss-soc-bus";
+			reg = <0x4a326000 0x2000>;
+			ti,hwmods = "pruss";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges;
+			status = "disabled";
+
+			pruss: pruss@4a300000 {
+				compatible = "ti,am3356-pruss";
+				reg = <0x4a300000 0x2000>,
+				      <0x4a302000 0x2000>,
+				      <0x4a310000 0x3000>,
+				      <0x4a326000 0x2000>,
+				      <0x4a32e000 0x31c>,
+				      <0x4a332000 0x58>;
+				reg-names = "dram0", "dram1", "shrdram2", "cfg",
+					    "iep", "mii_rt";
+				#address-cells = <1>;
+				#size-cells = <1>;
+				ranges;
+				status = "disabled";
+
+				pruss_intc: intc@4a320000 {
+					compatible = "ti,am3356-pruss-intc";
+					reg = <0x4a320000 0x2000>;
+					reg-names = "intc";
+					interrupts = <20 21 22 23 24 25 26 27>;
+					interrupt-names = "host2", "host3",
+							  "host4", "host5",
+							  "host6", "host7",
+							  "host8", "host9";
+					interrupt-controller;
+					#interrupt-cells = <1>;
+				};
+
+				pru0: pru@4a334000 {
+					compatible = "ti,am3356-pru";
+					reg = <0x4a334000 0x2000>,
+					      <0x4a322000 0x400>,
+					      <0x4a322400 0x100>;
+					reg-names = "iram", "control", "debug";
+					interrupt-parent = <&pruss_intc>;
+					interrupts = <16>, <17>;
+					interrupt-names = "vring", "kick";
+					status = "disabled";
+				};
+
+				pru1: pru@4a338000 {
+					compatible = "ti,am3356-pru";
+					reg = <0x4a338000 0x2000>,
+					      <0x4a324000 0x400>,
+					      <0x4a324400 0x100>;
+					reg-names = "iram", "control", "debug";
+					interrupt-parent = <&pruss_intc>;
+					interrupts = <18>, <19>;
+					interrupt-names = "vring", "kick";
+					status = "disabled";
+				};
+			};
+		};
+
 		elm: elm@48080000 {
 			compatible = "ti,am3352-elm";
 			reg = <0x48080000 0x2000>;
-- 
1.7.9.5

